indigo_api/models/works.py

```diff
@@ -359,6 +362,55 @@ def as_at_date(self):
         if dates:
             return max(dates)
 
+    def all_provisions(self):
```

Comment:
I suggest this function simply finds the first expression and then delegates to the same method on the expression document, which then introspects the TOC etc. Provisions are inherently tied to a point in time.